traits/observers/expression.py

```diff
@@ -196,11 +276,88 @@ def join_(*expressions):
     return _functools.reduce(lambda e1, e2: e1.then(e2), expressions)
 
 
+def dict_items(notify=True, optional=False):
+    """ Create a new expression for observing items inside a dict.
+
+    Events emitted (if any) will be instances of
+    :py:class:`~traits.observers.events.DictChangeEvent`.
+
+    If an expression with ``dict_items`` is further extended, the
+    **values** of the dict will be given to the next item in the expression.
+    For example, the following observes a trait named "number" on any object
+    that is one of the values in the dict named "mapping"::
+
+        trait("mapping").dict_items().trait("number")
+
+    Parameters
+    ----------
+    notify : boolean, optional
+        Whether to notify for changes.
+    optional : booleal, optional
+        Whether to ignore this if the upstream object is not a dict.
+
+    Returns
+    -------
+    new_expression : traits.observers.expressions.Expression
+    """
+    observer = _DictItemObserver(notify=notify, optional=optional)
+    return SingleObserverExpression(observer)
+
+
+def list_items(notify=True, optional=False):
+    """ Create a new expression for observing items inside a list.
+
+    Events emitted (if any) will be instances of
+    :py:class:`~traits.observers.events.ListChangeEvent`.
+
+    e.g. ``trait("containers").list_items()`` for observing to mutations
+    to a list named ``containers``.
+
+    e.g. ``trait("containers").list_items().trait("value")`` for observing
+    the trait ``value`` on any items in the list ``containers``.
+
+    Parameters
+    ----------
+    notify : boolean, optional
+        Whether to notify for changes.
+    optional : booleal, optional
+        Whether to ignore this if the upstream object is not a list.
+
+    Returns
+    -------
+    new_expression : traits.observers.expressions.Expression
+    """
+    observer = _ListItemObserver(notify=notify, optional=optional)
+    return SingleObserverExpression(observer)
+
+
+def set_items(notify=True, optional=False):
+    """ Create a new expression for observing items inside a set.
+
+    Events emitted (if any) will be instances of
+    :py:class:`~traits.observers.events.SetChangeEvent`.
+
+    Parameters
+    ----------
+    notify : boolean, optional
+        Whether to notify for changes.
+    optional : booleal, optional
+        Whether to ignore this if the upstream object is not a set.
+
+    Returns
+    -------
+    new_expression : traits.observers.expressions.Expression
```

Comment:
Oops, this `"expressions"` needs to be changed to `"expression"`.

Reply:
I see a few cases of `expressions` in this module, and one in `observe.py`.

Reply:
Thanks!  It is on my list for a final sweep of cleanups.